libs/MeshKernel/src/Mesh2DIntersections.cpp

```diff
@@ -0,0 +1,363 @@
+            m_mesh,
+            polyLine,
+            vistedEdges);
+
+        // no valid seed found in the entire mesh, we are done
+        if (!isSeedFound)
+        {
+            break;
+        }
+
+        const auto crossingNextSegmentIndex = crossedSegmentIndex + 1;
+        crossingEdges.push({crossedEdgeIndex, crossedSegmentIndex, crossingNextSegmentIndex});
+
+        // use breadth search along the current polyline
+        while (!crossingEdges.empty())
+        {
+            auto [currentCrossingEdge, segmentIndex, nextSegmentIndex] = crossingEdges.front();
+            crossingEdges.pop();
+
+            for (UInt f = 0; f < m_mesh.m_edgesFaces[currentCrossingEdge].size(); ++f)
+            {
+                const auto currentFaceIndex = m_mesh.m_edgesFaces[currentCrossingEdge][f];
+
+                if (currentFaceIndex == constants::missing::uintValue)
+                {
+                    continue;
+                }
+
+                for (UInt e = 0; e < m_mesh.m_facesEdges[currentFaceIndex].size(); ++e)
+                {
+                    const auto edgeIndex = m_mesh.m_facesEdges[currentFaceIndex][e];
+                    if (vistedEdges[edgeIndex] && vistedFace[currentFaceIndex])
+                    {
+                        continue;
+                    }
+
+                    UInt firstIndex = segmentIndex;
+                    UInt secondIndex = nextSegmentIndex;
+
+                    auto intersection = AreSegmentsCrossing(polyLine[firstIndex],
+                                                            polyLine[secondIndex],
+                                                            m_mesh.m_nodes[m_mesh.m_edges[edgeIndex].first],
+                                                            m_mesh.m_nodes[m_mesh.m_edges[edgeIndex].second],
+                                                            false,
+                                                            m_mesh.m_projection);
+
+                    bool intersectionFound = std::get<0>(intersection);
+                    double crossProductValue = std::get<2>(intersection);
+                    double adimensionalPolylineSegmentDistance = std::get<3>(intersection);
+                    double adimensionalEdgeDistance = std::get<4>(intersection);
+
+                    if (!intersectionFound)
+                    {
+                        const auto segmentIntersection = GetNextEdgeIntersection(polyLine, edgeIndex, segmentIndex, nextSegmentIndex, 1);
+                        intersectionFound = std::get<0>(segmentIntersection);
+                        firstIndex = std::get<1>(segmentIntersection);
+                        secondIndex = std::get<2>(segmentIntersection);
+                        crossProductValue = std::get<3>(segmentIntersection);
+                        adimensionalPolylineSegmentDistance = std::get<4>(segmentIntersection);
+                        adimensionalEdgeDistance = std::get<5>(segmentIntersection);
```

Comment:
Could these be put in a struct? this could then be returned from tthe GetNextEdgeIntersection function and the same struct could be passed to the updateEdgeIntersections function to reduce the number of parameters